pants.toml

```diff
@@ -159,6 +159,16 @@ extra_env_vars = [
   "RUST_BACKTRACE=1",
 ]
 
+[mypy]
+interpreter_constraints = [">=3.7,<3.10"]
+extra_requirements.add = [
+  "mypy-typing-asserts",
+]
+extra_type_stubs.add = [
+  "mypy-typing-asserts",
+]
```

Comment:
We have to add it to both:
- `requirements` so that it's in the checker pex
- `extra_type_stubs` so that its in the checkee pex (because it will be imported)

Reply:
It's correct to put in `extra_requirements` because we need that for `mypy.pex` to load the plugin.

But instead of using `extra_type_stubs`, it's better to use a normal `python_requirement` target, like put it in `requirements.txt`. That's because we import it in our actual code, so we want to model it properly so things like `./pants dependencies` work.

It's not great that we have to duplicate the requirement twice. https://github.com/pantsbuild/pants/issues/12449 would address that.

Reply:
Yeah it's a shame we have to put it in both :sob: I wonder if there's a way to have the mypy plugin function trigger for non-qualified function calls. I'll try that.

Otherwise I'll put it in `requirements.txt`, but that feels a bit shameful :disappointed:

Reply:
Yeah I think you want both requirements.txt + `[mypy].extra_requirements`. Agreed it's a shame.

Reply:
updated